SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerImpl.java

```diff
@@ -153,6 +153,9 @@ public void subscribed(Subscribed subscribed) {
         heartbeatIntervalSeconds = Optional.of(advertisedHeartbeatIntervalSeconds);
       }
 
+      // Should be called before activation of leader cache or cache could be left empty
+      startup.checkMigrations();
```

Comment:
Do we expect that this wouldn't have run at startup?

Reply:
It only runs when we take leadership, not as a Managed class. Moved it here since we need to migrate before filling leader cache, since leader cache is using the newer taskManager at this point